pkg/downloader/chart_downloader.go

```diff
@@ -158,6 +166,52 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+func (c *ChartDownloader) DownloadToInMemoryCache(ref, version string) (string, *provenance.Verification, error) {
+	u, err := c.ResolveChartVersion(ref, version)
+	if err != nil {
+		return "", nil, err
+	}
+
+	g, err := c.Getters.ByScheme(u.Scheme)
+	if err != nil {
+		return "", nil, err
+	}
+
+	chartData, err := g.Get(u.String(), c.Options...)
+	if err != nil {
+		return "", nil, err
+	}
+
+	// If provenance is requested, verify it.
+	ver := &provenance.Verification{}
+	if c.Verify > VerifyNever {
+		provData, err := g.Get(u.String() + ".prov")
+		if err != nil {
+			if c.Verify == VerifyAlways {
+				return ref, ver, errors.Errorf("failed to fetch provenance %q", u.String()+".prov")
+			}
+			fmt.Fprintf(c.Out, "WARNING: Verification not found for %s: %s\n", ref, err)
+			return ref, ver, nil
+		}
+
+		if c.Verify != VerifyLater {
+			ver, err = VerifyInMemoryChart(ref, c.Keyring, chartData, provData)
+			if err != nil {
+				// Fail always in this case, since it means the verification step
+				// failed.
+				return ref, ver, err
+			}
+		}
+	}
+
+	InMemoryCache[ref] = InMemoryChart{
```

Comment:
If we really need a map, then don't we need to index on both ref and version? Or else a request for the same chart but a different version will overwrite the previous request of the same chart. If this is not a problem, maybe we don't need a map at all?

Reply:
I have added the Version attribute for the chart. I think we don't really need it as of now, but for the sake of semantics we should keep it. Otherwise what is tracking the InMemory Charts ?

Reply:
the InMemoryChart would need to be used by the API as this is thestore where the charts are kept and not in the filesystem.

Reply:
Sorry, I wasn't clear.  

When downloading dependencies, is it possible that we will need to download multiple versions of the same chart name?  I don't know enough about dependencies to answer that.  We need another maintainer.

I'm asking because, in the code
```
InMemoryCache[ref] = InMemoryChart{
```
the variable `ref` is just the chart name (does not include the version I believe).  So, if we need to download multiple versions of the same chart name, then the index of `InMemoryCache` needs to contain both  the name and the version of the chart.

Say we download `stable/grafana` version 1.6 and then `stable/grafana` version 1.2.  The second download will replace the first since the same name is used as `ref`.

Reply:
That makes sense, the ref should be the name and the version of the chart. But considering that in the code the chart is read solely based on the name without the version so I guess installing multiple versions of the same chart or having them as dependencies is not a scenario considered here.